stack/stack.py

```diff
@@ -149,19 +149,37 @@ def __init__(self, scope: Construct, id: str, **kwargs) -> None:
             self, "landsat_output_bucket", OUTPUT_BUCKET
         )
 
+        sentinel_input_bucket_expiration_days = int(
+            os.environ["HLS_SENTINEL_INPUT_BUCKET_EXPIRATION_DAYS"]
+        )
+
         # Must be created as part of the stack due to trigger requirements
         self.sentinel_input_bucket = aws_s3.Bucket(
             self,
             "SentinelInputBucket",
             bucket_name=SENTINEL_INPUT_BUCKET,
             removal_policy=RemovalPolicy.DESTROY,
+            lifecycle_rules=[
+                aws_s3.LifecycleRule(
```

Comment:
Feel free to disregard, but I was wondering if we also want to add a policy to "abort incomplete multipart uploads"?

There's an option `abort_incomplete_multipart_upload_after: Optional[Duration]`,
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3/LifecycleRule.html

The objects in these buckets are large enough that they're probably multi-part uploads, and if one fails (e.g., the uploading worker dies for some reason) the upload will be cancelled but the storage still lives on (costing $$),
https://repost.aws/knowledge-center/s3-incomplete-multipart-uploads

I remember this saved us a decent chunk of change at the last place I worked

Reply:
Good call. I added that to all the buckets with lifecycle rules. Incomplete multipart uploads will now be aborted after 1 day.